### DISCUSSION TITLE
Variable constraints based on optimization variables

### DISCUSSION THREAD

**Discussion by LouisThomasRoy** [casadi/casadi#3537] [Q&A]
Hi, I am building a complex vehicle simulation where one of my boundary constraints depends on one of my optimization variables. How can I implement variable constraints based on one of the optimization variables, if it is even possible? I have tried to replicate the issue on a more simple problem, such as the following where y is bounded by x:

```import casadi.*
 
% Problem: 
%     minimize    x^2 + y^2
%     subject to  5*x - 8*y + 8 == 0 and 2*x <= y <= 3*x
x = SX.sym('x');
y = SX.sym('y');
v = [x;y];
f = x^2 + y^2;
g = 5*x - 8*y + 8;
g = [g y];
nlp = struct('x', v, 'f', f', 'g', g);

% Create IPOPT solver object
solver = nlpsol('solver', 'ipopt', nlp);

% Solve the NLP
res = solver('x0' , [2.5 3.0],... % solution guess
             'lbx', [-inf -inf],...           % lower bound on x
             'ubx',  [inf inf],...           % upper bound on x
             'lbg',    [0 2*x],...           % lower bound on g
             'ubg',    [0 3*x]);             % upper bound on g
 
x_opt = full(res.x);
disp(x_opt)```

**Comment by jgillis** [comment]
You must rewrite these constraints such that the bounds are numeric.

e.g 2<=y/x<=3
or 2x-y<=0 and 3x-y>=0

The latter is probably preferrable because of linearity
